### ISSUE TITLE
Always 403

### ISSUE THREAD

**Issue by xumingxiang** [Tmeister/wp-api-jwt-auth#78]
Always 403
my wp version is 4.7.5 
my site url  is http://localhost:8080/wordpress/
the .htaccess content is 

# BEGIN WordPress
<IfModule mod_rewrite.c>
RewriteEngine On
RewriteCond %{HTTP:Authorization} ^(._) 
RewriteRule ^(._) - [E=HTTP_AUTHORIZATION:%1] 

RewriteEngine On
RewriteBase /wordpress/
RewriteRule ^index\.php$ - [L]
RewriteCond %{REQUEST_FILENAME} !-f
RewriteCond %{REQUEST_FILENAME} !-d
RewriteRule . /wordpress/index.php [L]
</IfModule>

# END WordPress
and  the wp-config.php 
........
define('JWT_AUTH_SECRET_KEY', 'ZkN`H=qj6DzfRz|cT-II_GW|1S1ovPfZse?[>mAc.v(.iEmGN$>8K`ou(P|Lkq]y');
define('JWT_AUTH_CORS_ENABLE', true);
.........

help me !!!!

**Comment by Orabig** [comment]
TL;DR;
Move the definition of JWT_AUTH_SECRET_KEY to the top of your wp-config.php file

Detailed solution :

As stated in the installation doc of the plugin,
the JWT needs a secret key to sign the token this secret key must be unique and never revealed.

Thus, you must add this line into your wp-config.php file

define('JWT_AUTH_SECRET_KEY', 'your-top-secrect-key');
However, what the doc does not say, is that you should not put this at the end of the file.
Insert this line just after the other definitions (AUTH_KEY, SECURE_AUTH_KEY, ... NONCE_SALT), and the 403 error should be gone.

I guess that's because the definition of JWT_AUTH_SECRET_KEY must be done before require_once(ABSPATH . 'wp-settings.php');

**Comment by tvler** [comment]
Hopefully this will help someone. I actually got a successful token using this js syntax:
```
fetch(`[your cool wp site]/wp-json/jwt-auth/v1/token`, {
   method: 'post',
   headers: {
      'Content-Type': 'application/x-www-form-urlencoded; charset=utf-8',
   },
   body: 'password=[your cool pw]&username=[your cool un]',
}).then(r => console.log(r));
```

**Comment by hassandiv** [comment]
You saved me today! thanks
